### DISCUSSION TITLE
Skia Gtk TextBox and Control keystrokes

### DISCUSSION THREAD

**Discussion by sarosigy** [unoplatform/uno#16792] [Q&A] [answered]
First of all, thank you for the brand new Skia rendering, it is a huge improvement compared to the previous implementation.

However, I have a slight problem which I would like to solve.
With the previous implementation, the following would 'swallow' the control characters:

```csharp
    public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
            KeyDown += (s, e) =>
            {
                var state = Windows.UI.Core.CoreWindow.GetForCurrentThread().GetKeyState(Windows.System.VirtualKey.Control);
                if ((state & Windows.UI.Core.CoreVirtualKeyStates.Down) == Windows.UI.Core.CoreVirtualKeyStates.Down)
                {
                    if (e.Key  == Windows.System.VirtualKey.S)
                    {
                        e.Handled = true;
                        SaveFile_Click(null, null);
                    }
                }
            };
        }
    }
```
And with on a TextBox put onto the Page, when the Ctrl-S is pressed, the keystroke is swallowed, the file saved.

Now, the following happens after pressing the Ctrl-S:
![image](https://github.com/unoplatform/uno/assets/37872626/be3ab7b8-7d7a-455a-953a-dd22928ba93b)
The SaveFile_Click is called properly, setting a breakpoint shows that the e.Handled is set to true.

Actually I tried this, also, but without success:
```csharp
        private void TextBox_OnKeyDown(object sender, KeyRoutedEventArgs e)
        {
            var state = Windows.UI.Core.CoreWindow.GetForCurrentThread().GetKeyState(Windows.System.VirtualKey.Control);
            if ((state & Windows.UI.Core.CoreVirtualKeyStates.Down) == Windows.UI.Core.CoreVirtualKeyStates.Down)
            {
                e.Handled = true;
            }
        }
```

How to properly handle the Ctrl-S key event without putting the letter 's' into the TextBox?

**Comment by spouliot** [comment]
That sounds like a bug.

Are you using the Gtk project (as the title suggests) or the new Skia `netXX-desktop` target ?

If the latter it would useful to know on which OS this happens since the low-level code (like keyboard) has OS specific code.

**Comment by sarosigy** [comment]
Okay, this is a legacy project updated to the latest Uno, so it uses the Gtk head.

I will report back soon.

**Reply by sarosigy** [reply, marked as answer]
I tried with a brand new project (dotnet new unoapp -platforms desktop) and it seems to be fine.
The OS is a Debian bookworm, X11.
